cardano-api/src/Cardano/Api/SerialiseRaw.hs

```diff
@@ -31,9 +32,27 @@ serialiseToRawBytesHex = Base16.encode . serialiseToRawBytes
 serialiseToRawBytesHexText :: SerialiseAsRawBytes a => a -> Text
 serialiseToRawBytesHexText = Text.decodeUtf8 . serialiseToRawBytesHex
 
-deserialiseFromRawBytesHex :: SerialiseAsRawBytes a
-                           => AsType a -> ByteString -> Maybe a
-deserialiseFromRawBytesHex proxy hex =
-    case Base16.decode hex of
-      Right raw -> deserialiseFromRawBytes proxy raw
-      Left _msg -> Nothing
+data RawBytesHexError
+  = RawBytesHexErrorBase16DecodeFail
+      ByteString -- ^ original input
+      String -- ^ error message
+  | RawBytesHexErrorRawBytesDecodeFail
+      ByteString -- ^ original input
+      -- TODO(2022-01-26, cblp) TypeRep -- ^ output type proxy
+  deriving (Show)
+
+instance Error RawBytesHexError where
+  displayError = \case
+    RawBytesHexErrorBase16DecodeFail input message ->
+      "Expected Base16-encoded bytestring, but got " ++ show input ++ "; "
+      ++ message
+    RawBytesHexErrorRawBytesDecodeFail input ->
+      "Failed to deserialise " ++ show input
+      -- TODO(2022-01-26, cblp) show expected output type
+
+deserialiseFromRawBytesHex ::
```

Comment:
If you have a look in the code base you'll see we usually indent type signatures that break the 100 char limit as follows:
```
deserialiseFromRawBytesHex 
  :: SerialiseAsRawBytes a 
  => AsType a 
  -> ByteString 
  -> Either RawBytesHexError
```
Although in this case because we are under 100 chars, we don't need any indentation necessarily. Either or is fine.

Reply:
I also found some files using Ormolu style, so I applied it here. Okay, looks like this file should be in classic style.

Reply:
The style guide says the (soft) limit is 80 chars.